sql/analyzer/pushdown_test.go

```diff
@@ -156,6 +156,55 @@ func TestPushdownFilterToTables(t *testing.T) {
 				),
 			),
 		},
+		{
+			name: "pushdown filter to aliased tables",
+			node: plan.NewProject(
+				[]sql.Expression{
+					expression.NewGetFieldWithTable(0, sql.Int32, "t1", "i", true),
+				},
+				plan.NewFilter(
+					and(
+						expression.NewEquals(
+							expression.NewGetFieldWithTable(1, sql.Float64, "t1", "f", false),
+							expression.NewLiteral(3.14, sql.Float64),
+						),
+						expression.NewIsNull(
+							expression.NewGetFieldWithTable(0, sql.Int32, "t2", "i2", false),
+						),
+					),
+					plan.NewCrossJoin(
+						plan.NewTableAlias("t1",
+							plan.NewResolvedTable(table, nil, nil)),
+						plan.NewTableAlias("t2",
+							plan.NewResolvedTable(table2, nil, nil)),
+					),
+				),
+			),
+			expected: plan.NewProject(
+				[]sql.Expression{
+					expression.NewGetFieldWithTable(0, sql.Int32, "t1", "i", true),
+				},
+				plan.NewCrossJoin(
+					plan.NewDecoratedNode("Filtered table access on [(t1.f = 3.14)]",
+						plan.NewTableAlias("t1",
+							plan.NewResolvedTable(table.WithFilters(sql.NewEmptyContext(),
+								[]sql.Expression{
+									expression.NewEquals(
```

Comment:
I verified this expected value matched the actual value, but it doesn't look like our tests actually verify the full depth of the expected output. To test that, I tried changing these deep values and the tests still passed correctly. In the console, I also noticed a message about "max depth reached", but I couldn't find any golang documentation about that; do you know if that's a known limitation with reflect's DeepEquals? 

It seems like the tests aren't comparing the full depth of the expected output and I'm wondering if there are other issues lurking.

Reply:
`table.WithFilter` copies the top-level Table but shares the slices between each. The last test sets the filters correctly, so that's the one this test also uses.

Reply:
Ah! Nice catch on the embedded *Table within FilteredTable. That makes the behavior make a lot more sense. I went ahead and added some helper methods to make separate objects and haven't been able to trigger the same problem since then.